Updated model architecture (#62)
thesis20/thesis_paper: sections/csgcn/csgcn-chapter.tex

With this intuition, we will now detail how this is implemented in the CSGCN-IS model.

\subsection{Model architecture}\label{subsec:csgcn_is_model_architecture}

> Based on the considerations made in \Cref{sec:foundational_knowledge}, the CSGCN models are based on these observations made in SGC \cite{SimplifyingGCN} and the LightGCN model \cite{LightGCN} about simplifying the models to a linear model.

Reviewer: havde vi defineret SGC før?

Author: Yes, i introen - det er i #54

To facilitate the model architecture, each node in the graph is represented by a low-dimensional vector, also called an embedding.
By doing this, the model can learn the nodes' properties and capture them in the embeddings.
The model consists of four parts: Input, graph convolution layers, layer combination, and a prediction function.\\
The input is represented as a quadripartite graph consisting of users, items, user side-information, and item side-information.

> This graph is used as the input for the graph convolution layers, which serves as a way to capture higher-order connectivities between the nodes in the graph.

Reviewer: ```suggestion
This graph is used as the input for the graph convolution layers, which serve to capture higher-order connectivities between the nodes in the graph.
```

> Each convolutional layer takes is stacked on top of the previous layer, such that each layer captures information from nodes that are further away from the central node.

Reviewer: ```suggestion
Each convolutional layer is stacked on top of the previous layer, such that each layer captures information from nodes that are further away from the central node.
```

> Having passed through $L$ convolution layers, we are left with an embedding representation of each node for each layer, which is then passed to the layer combination function, which takes a weighted sum of each representation to generate a final representation for each node.\\

Reviewer: ```suggestion
Having passed through $L$ convolution layers, we are left with an embedding representation of each node for each layer, which is then passed to the layer combination function, taking a weighted sum of each representation to generate a final representation for each node.\\
```

Finally, these combined representations of users and items are passed to a prediction function along with a context embedding to calculate a score for the given user, item, and context combination as described in \Cref{subsec:csgcn_is_intuition}.\\

> These predicted scores are then used to generate a top-$k$ list of recommended items for a user in the given context.\\

Reviewer: ```suggestion
These predicted scores are used to generate a top-$k$ list of recommended items for a user in the given context.\\
```
